docs/design/features/native-hosting.md

```diff
@@ -333,17 +399,61 @@ Calling this function will load the specified assembly in isolation (into its ow
 * `type_name` - Assembly qualified type name to find
 * `method_name` - Name of the method on the `type_name` to find. The method must be `static` and must match the signature of `delegate_type_name`.
 * `delegate_type_name` - Assembly qualified delegate type name for the method signature, or null. If this is null, the method signature is assumed to be:
-```C#
-public delegate int ComponentEntryPoint(IntPtr args, int sizeBytes);
-```
-This maps to native signature:
+    ```C#
+    public delegate int ComponentEntryPoint(IntPtr args, int sizeBytes);
+    ```
+    This maps to native signature:
+    ```C
+    int component_entry_point_fn(void *arg, int32_t arg_size_in_bytes);
+    ```
+    **[.NET 5 and above]** The `delegate_type_name` can be also specified as `UNMANAGEDCALLERSONLY_METHOD` (defined as `(const char_t*)-1`) which means that the managed method is marked with `UnmanagedCallersOnlyAttribute`.
+* `reserved` - parameter reserved for future extensibility, currently unused and must be `NULL`.
+* `delegate` - out parameter which receives the native function pointer to the requested managed method.
+
+The helper will always load the assembly into an isolated load context. This is the case regardless if the requested assembly is also available in the default load context or not.
+
+**[.NET 5 and above]** It is allowed to call this helper on a host context which came from `hostfxr_initialize_for_dotnet_command_line` which will make all the application assemblies available in the default load context. In such case it is recommended to only use this helper for loading plugins external to the application. Using the helper to load assembly from the application itself will lead to duplicate copies of assemblies and duplicate types.
```

Comment:
Do we no longer want to also relax this to allow `NULL` for the host context to be the first host context?

Reply:
I removed it to make the change simpler.
I'm fine adding that functionality as well - but I think it should be a separate change (both to the doc and to the implementation).